streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -34,6 +34,8 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.Collection;
+import java.util.stream.Collectors;
```

Comment:
I cannot comment below, but I think we should still simplify the code when the user specifies a partition. I left a comment on the original PR about it (https://github.com/apache/kafka/pull/7984/files#r375513526). Let me know if this comment makes sense.

Reply:
Thanks @mjsax . It makes sense, I have made the suggested change.